src/jit/codegenlegacy.cpp

```diff
@@ -5426,6 +5426,9 @@ void CodeGen::genCodeForTreeLeaf_GT_JMP(GenTreePtr tree)
             offset += REGSIZE_BYTES;
         }
     }
+
+    // Lock the argument registers s.t. they are not trashed by code that runs before the jmp (e.g. GS cookie checks).
+    regSet.rsLockReg(fixedArgsMask);
```

Comment:
> as they mark all arg registers as in-use before generating the GS check and free them after generating the call

That doesn't matter, because rsGrabReg() might spill an in-use register.

Why do you prefer your solution instead of mine, which isolates the change to the GS cookie code, and doesn't require a global variable?